File: leet_easy_c++/leet_066/leet_066a.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        int len = digits.size();
        int cur = len;
        vector<int> ans(1,1);
        if(len == 0) return ans;
        ans = digits;
        ans[cur-1]++;
        while(cur > 0 && ans[cur-1]==10){
            ans[cur-1]=0;
            if(cur-1>0) ans[cur-2]++;
            else{
                vector<int> ans2(len+1,0);
                ans2[0]=1;
                return ans2;
            }
            cur--;
        }
        return ans;
    }
};

File: leet_easy_python/leet_066/leet_066a.py
class Solution(object):
    def plusOne(self, digits):
        j = len(digits)-1
        digits[j]+=1
        while j>0 and digits[j]==10:
            digits[j]=0
            digits[j-1]+=1
            j-=1
        if j==0 and digits[j] == 10:
            digits[0] = 0
            digits.insert(0,1)
        return digits
        
        """
        :type digits: List[int]
        :rtype: List[int]
        """
        